### DISCUSSION TITLE
Optimized editing interface for content translation

### DISCUSSION THREAD

**Discussion by zhuyifang** [directus/directus#8297] [Feature Requests]
Using this tab method for multi-language management will be more convenient to operate and improve user experience
![image](https://user-images.githubusercontent.com/16346219/134711675-172bf08a-a78e-4c28-8c8f-cff39aa33a43.png)

**Comment by benhaynes** [comment]
Interesting, but that is set per field... making it hard to switch a whole form to a new language. Also, this works with only a few languages on a field that is full-width — if you have many languages, and half-width fields, then this will wrap and quickly become cumbersome.

In any event, the Core features are meant to cover 80%+ of the user needs, but if you want to create more custom interfaces, we have built the entire platform with modular extensibility. Check out the Interface Extensions. https://docs.directus.io/guides/interfaces/

**Reply by zhuyifang** [reply]
Actually I don't agree with you.
I think it does not matter whether this is full-width or not.
for example:
We have configured 3 languages for field A
Then in the form, the display of the field
A default lang1 lang2 lang3
Then enter the values of different languages by switching tabs.

**Reply by benhaynes** [reply]
I think you misunderstand my comment. What happens when there are more translations, and they don't fit in the width of the field? This gets messy very fast with 10+ languages.

**Reply by zhuyifang** [reply]
It’s fine if the tab can be scrolled left and right, and there are usually not so many languages

**Reply by benhaynes** [reply]
Scrolling would make it very annoying to get to the last language, especially if you use that a lot. Also, "usually not so many languages" is a big assumption based on your use-case though. 😉

**Reply by zhuyifang** [reply]
There is another approach in interaction design
That is, when a row is not displayed completely, a drop-down box is displayed at the end. In this case, his interaction is basically the same as the existing one, but the form will be more compact

**Reply by zhuyifang** [reply]
But in general, it will be more convenient to use

**Reply by zhuyifang** [reply]
![image](https://user-images.githubusercontent.com/16346219/134908085-61ed658c-6014-4fb8-b2e4-10de550a4df0.png)
like this

**Comment by zhuyifang** [comment]
In addition, in this function, I found one thing:
When using automatic setting by content translation, the fields set as required are not automatically set, but new fields need to be added manually.
Don't know if this is a bug

**Reply by benhaynes** [reply]
Bugs should be reported separately, since this is a feature request.

**Reply by zhuyifang** [reply]
Yeah, I’m just not sure if this is a bug, so I brought it up while discussing this issue

**Reply by joselcvarela** [reply]
Doesn't seem to be a bug.
When you click on create translations, it automatically create a table `languages` and insert some languages by default. It also creates the relationship between your model and translations.

But, it doesn't create any fields. That part should be the user.

**Reply by zhuyifang** [reply]
But here, he said: All required fields and relationships will be set automatically

But it seems, in fact, there is no, is my understanding of this sentence wrong?
![image](https://user-images.githubusercontent.com/16346219/134751452-9ab5288e-40c8-4921-ae53-597930c6af73.png)

**Reply by joselcvarela** [reply]
The meaning of this message is it will create a table `languages` and a table `[your_model]_translations`.
Then it also will create the columns `code` and `name` in table `languages` and `id` , `[your_model]_id` and `languages_id` on table `[your_model]_translations`.

So the `required fields` parts refers to the columns mentioned above.
But, yeah, we could probably change the message. Do you have any suggestions? 🤔

**Reply by zhuyifang** [reply]
If like you said, I think it can be like this:
1. While maintaining the existing interface interaction:
Adjust the description sentence, at least in the Chinese language environment, it seems to be wrong. My understanding is that he will automatically create the model fields that need to be translated, and all the model fields I set as mandatory.

2. Optimize the existing design:
When creating a translation, you can allow users to check the fields that need to be translated, and then the corresponding ones are automatically created in the relationship table (using the same field type and configuration)

**Reply by joselcvarela** [reply]
1. You can contribute to improve our translations. Check here https://docs.directus.io/getting-started/backing-directus/#translations

2. It actually could be a good idea. But you are assuming you will have fields outside translations. Why not put all the translatable fields inside translations? I understand if a translation is missing you want go to the default, but why not rely on a default language? By doing this way, you would benefit of the translations interface like percentage of what is translated.
Also, you could make required fields per language using Conditions (https://docs.directus.io/guides/fields/#conditions)

![image](https://user-images.githubusercontent.com/14039341/134795624-247c3bb3-8455-47f1-ab1a-0a88dfcde9ec.png)

**Reply by zhuyifang** [reply]
In fact, I tried to contribute to the Chinese translation,
But I have no confidence in the quality of my contribution. The main reason is that the grammatical structure of Chinese and English is different. Then here: https://locales.directus.io/, which is stripped into words in many places. As a directus who has been using @76 to @94, it is difficult to understand, so I will translate it individually.

If the Chinese translation is only through 2 words, it will be a readable but weird translation.

So I just mentioned a RP before. The system can provide more information, such as screenshots of the interface.

Regarding your second point, I think it may be that I did not describe it clearly, or you misunderstood my previous reply.

For automatically created translation, if it can be like this in the picture, then it is the real automatic translation function.
![image](https://user-images.githubusercontent.com/16346219/134796182-7bb0122e-1a5d-4769-8030-12bdea232ec4.png)

**Reply by joselcvarela** [reply]
I did understand your second point, but you are putting fields with checkboxes, so you are already assuming that those fields already exists. But if translations do not exist yet, it means the fields are in model. This could misleads the final user since we will need to show every field on the model, like `id`, `user_created`, `date_created`, etc.

In my point of view, you should create the translations first, then add the translatable fields inside translations. By doing this way, you only have translatable fields inside translations and not outside which will be more organised. Also, you can take advantage of Conditionals to make translatable fields required for specific languages.

**Reply by zhuyifang** [reply]
how about this? how do you feel
![image](https://user-images.githubusercontent.com/16346219/134850023-8080a738-a9aa-41be-b01b-0052a48b244f.png)

**Reply by benhaynes** [reply]
Translations can't and won't exist on the parent table, so we won't show them as if they're on the parent table since that would confuse developers in where the data exists, and should be fetched within the API. It needs to be clear that translations are in a relational table, and we're working on a nested relational review for fields in the data model.

Similarly, there won't be a button to convert normal fields to translations individually. Once you understand how the system works now, it's easy enough to adapt your workflow. There's no reason to add additional code complexity to cover converting fields.

**Reply by zhuyifang** [reply]
The field 1 and field 2 that I have identified on the picture actually mean: in the model, which fields need to be translated. If you check it, the field will be automatically created in the translation table

**Reply by joselcvarela** [reply]
I already were there @zhuyifang but like @benhaynes says, you should not have translatable fields on the parent. All translatable fields should go into translation. By doing this way, your interface and your queries will be cleaner than having the same field in two different places.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
